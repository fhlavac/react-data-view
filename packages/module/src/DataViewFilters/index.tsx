export { default } from './DataViewFilters';
export * from './DataViewFilters';
