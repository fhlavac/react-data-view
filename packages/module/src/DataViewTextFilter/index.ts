export { default } from './DataViewTextFilter';
export * from './DataViewTextFilter';
