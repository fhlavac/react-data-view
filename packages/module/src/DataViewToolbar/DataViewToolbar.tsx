import React, { PropsWithChildren } from 'react';
import { Toolbar, ToolbarContent, ToolbarItem, ToolbarItemVariant } from '@patternfly/react-core';

export interface DataViewToolbarProps extends PropsWithChildren {
  /** Toolbar className */
  className?: string;
  /** Custom OUIA ID */
  ouiaId?: string;
  /** React component to display bulk select */
  bulkSelect?: React.ReactNode;
  /** React component to display pagination */
  pagination?: React.ReactNode;
  /** React component to display actions */
  actions?: React.ReactNode;
}

export const DataViewToolbar: React.FC<DataViewToolbarProps> = ({ className, ouiaId = 'DataViewToolbar', bulkSelect, actions = null, pagination, children, ...props }: DataViewToolbarProps) => (
  <Toolbar ouiaId={ouiaId} className={className} {...props}>
    <ToolbarContent>
      {bulkSelect && (
        <ToolbarItem data-ouia-component-id={`${ouiaId}-bulk-select`}>
          {bulkSelect}
        </ToolbarItem>
      )}
      {actions}
      {pagination && (
        <ToolbarItem variant={ToolbarItemVariant.pagination} data-ouia-component-id={`${ouiaId}-pagination`}>
          {pagination}
        </ToolbarItem>
      )}
      {children}
    </ToolbarContent>
  </Toolbar>
);

export default DataViewToolbar;

