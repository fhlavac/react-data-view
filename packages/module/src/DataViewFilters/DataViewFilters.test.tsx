import React from 'react';
import { render } from '@testing-library/react';
import DataViewFilters from './DataViewFilters';
import DataViewToolbar from '../DataViewToolbar';
import DataViewTextFilter from '../DataViewTextFilter';

describe('DataViewFilters component', () => {
  const mockOnChange = jest.fn();

  it('should render correctly', () => {
    const { container } = render(<DataViewToolbar
      filters={
        <DataViewFilters onChange={mockOnChange} values={{}}>
          <DataViewTextFilter filterId="one" title="One" />
          <DataViewTextFilter filterId="two" title="Two" />
        </DataViewFilters>
      }
    />);
    expect(container).toMatchSnapshot();
  });
});
