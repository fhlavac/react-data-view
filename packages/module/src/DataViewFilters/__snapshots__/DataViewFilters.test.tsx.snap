// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataViewFilters component should render correctly 1`] = `
<div>
  <div
    class="pf-v6-c-toolbar"
    data-ouia-component-id="DataViewToolbar"
    data-ouia-component-type="PF6/Toolbar"
    data-ouia-safe="true"
    id="pf-random-id-0"
  >
    <div
      class="pf-v6-c-toolbar__content"
    >
      <div
        class="pf-v6-c-toolbar__content-section"
      >
        <div
          class="pf-v6-c-toolbar__item"
        >
          <div
            class="pf-v6-c-toolbar__group pf-m-toggle-group pf-m-show-on-xl"
            data-ouia-component-id="DataViewFilters"
          >
            <div
              class="pf-v6-c-toolbar__toggle"
            >
              <button
                aria-disabled="false"
                aria-haspopup="false"
                aria-label="Show Filters"
                class="pf-v6-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF6/Button"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-v6-c-button__icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="pf-v6-c-toolbar__group pf-m-filter-group"
            >
              <div>
                <button
                  aria-expanded="false"
                  class="pf-v6-c-menu-toggle"
                  data-ouia-component-id="OUIA-Generated-MenuToggle-1"
                  data-ouia-component-type="PF6/MenuToggle"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-menu-toggle__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
                      />
                    </svg>
                  </span>
                  <span
                    class="pf-v6-c-menu-toggle__text"
                  >
                    One
                  </span>
                  <span
                    class="pf-v6-c-menu-toggle__controls"
                  >
                    <span
                      class="pf-v6-c-menu-toggle__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v6-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
              <div
                class="pf-v6-c-toolbar__item"
                data-ouia-component-id="DataViewTextFilter"
              >
                <div
                  class="pf-v6-c-text-input-group"
                  data-ouia-component-id="DataViewTextFilter-input"
                >
                  <div
                    class="pf-v6-c-text-input-group__main pf-m-icon"
                  >
                    <span
                      class="pf-v6-c-text-input-group__text"
                    >
                      <span
                        class="pf-v6-c-text-input-group__icon"
                      >
                        <svg
                          aria-hidden="true"
                          class="pf-v6-svg"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                          />
                        </svg>
                      </span>
                      <input
                        aria-label="One filter"
                        class="pf-v6-c-text-input-group__text-input"
                        id="one"
                        placeholder="Filter by One"
                        type="text"
                        value=""
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-c-toolbar__content pf-m-hidden"
      hidden=""
    >
      <div
        class="pf-v6-c-toolbar__group"
      />
      <div
        class="pf-v6-c-toolbar__group pf-m-action-group-inline"
      >
        <div
          class="pf-v6-c-toolbar__item"
        >
          <button
            aria-disabled="false"
            class="pf-v6-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="DataViewToolbar-clear-all-filters"
            data-ouia-component-type="PF6/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-v6-c-button__text"
            >
              Clear filters
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
